manim/utils/commands.py

```diff
@@ -27,21 +28,25 @@ def capture(command, cwd=None, command_input=None):
 
 
 def get_video_metadata(path_to_video: str | os.PathLike) -> dict[str]:
-    command = [
-        "ffprobe",
-        "-v",
-        "error",
-        "-select_streams",
-        "v:0",
-        "-show_entries",
-        "stream=width,height,nb_frames,duration,avg_frame_rate,codec_name",
-        "-print_format",
-        "json",
-        str(path_to_video),
-    ]
-    config, err, exitcode = capture(command)
-    assert exitcode == 0, f"FFprobe error: {err}"
-    return json.loads(config)["streams"][0]
+    with av.open(str(path_to_video)) as container:
+        stream = container.streams.video[0]
+        ctxt = stream.codec_context
+        rate = stream.average_rate
+        if stream.duration is not None:
```

Comment:
Yes, for .webm (VP9-encoded) files I had to implement this specific workaround.

Reply:
Ok Nice! I cannot find (only did a quick Google search) information backing up why VP9 would not include that information, but, anyway, I think this might be worth putting a small comment saying that it was observed for some formats, for example webm, that this field is None, so this information is not lost :)

Reply:
Ok nvm I found a page documenting that: https://trac.ffmpeg.org/wiki/FFprobeTips#Streamduration